### PR TITLE
Implementation of BehaviorMIQPAgent

### PR DESCRIPTION
**Pull request by klemense1**
- integration of MIQP planner
- Bug fixes in Map Generation from XODR 
- double instead of for Eigen Matrices

### CODE REVIEW

File: bark/world/opendrive/plan_view.cpp [hunk 1]
```diff
@@ -16,18 +16,32 @@ namespace bark {
...
 namespace opendrive {
 
-bool PlanView::AddLine(Point2d start_point, float heading, float length) {
+bool PlanView::AddLine(Point2d start_point, float heading, float length,
+                       float s_inc) {
   namespace bg = boost::geometry;
   using bark::geometry::Line;
   using bark::geometry::Point2d;
 
+  int num_points = length / s_inc;
```
Comment (patrickhart) [inline bark/world/opendrive/plan_view.cpp:25]: maybe add some documentation?

**Review by patrickhart — APPROVED** [review]
Looks good to me! See comments.

File: bark/geometry/commons.hpp [hunk 2]
```diff
@@ -31,7 +31,7 @@ namespace bg = boost::geometry;
 //! points
 template <typename T>
 using Point2d_t = bg::model::point<T, 2, bg::cs::cartesian>;
-using Point2d = Point2d_t<float>;
+using Point2d = Point2d_t<double>;
```
[the remark this answers is not in the document]
Comment (klemense1) [reply bark/geometry/commons.hpp:34]: Tobias changed all geometry containers (Point, Line, Polygon) to double...

File: bark/world/opendrive/plan_view.cpp (hunk 1 above)
Comment (klemense1) [reply bark/world/opendrive/plan_view.cpp:34]: alright, I added the doc

File: bark/world/map/map_interface.cpp [hunk 3]
```diff
@@ -174,12 +175,19 @@ void MapInterface::CalculateLaneCorridors(RoadCorridorPtr& road_corridor,
...
       if (next_lane == nullptr) break;
-      lane_corridor->GetCenterLine().ConcatenateLinestring(
-          next_lane->GetCenterLine());
-      lane_corridor->GetLeftBoundary().ConcatenateLinestring(
-          next_lane->GetLeftBoundary().line_);
-      lane_corridor->GetRightBoundary().ConcatenateLinestring(
+      Line new_center = bark::geometry::ConcatenateLinestring(
+          lane_corridor->GetCenterLine(), next_lane->GetCenterLine());
+      lane_corridor->SetCenterLine(new_center);
+      lane_corridor->SetFineCenterLine(new_center);
```
[the remark this answers is not in the document]
Comment (klemense1) [reply bark/world/map/map_interface.cpp:181]: yes. I guess it does not generate much overhead to store it, and all bark functions do not depend on it (only planners, possibly)

[the remark this answers is not in the document]
Comment (patrickhart) [reply bark/world/map/map_interface.cpp:181]: Okay, seems fine for me!

File: bark/benchmark/benchmark_result.py [hunk 4]
```diff
@@ -39,28 +39,35 @@ def configs_from_dict(behavior_dict):
...
-                 scenario_set_param_desc=None):
+                 scenario_set_param_desc=None, track_id_ego=None, 
+                 track_file_name=None):
         self.config_idx = config_idx
         self.behavior_config = behavior_config
         self.scenario = scenario
         self.scenario_idx = scenario_idx
         self.scenario_set_name = scenario_set_name
         self.scenario_set_param_desc = scenario_set_param_desc or {}
+        self.track_id_ego = track_id_ego
```
Comment (juloberno) [inline bark/benchmark/benchmark_result.py:50]: What is Data-scenario generation dependent should become part of the scenario set param description, which was introduced for the purpose of maintaining ScenarioGeneration specific additional infos.

File: bark/benchmark/benchmark_result.py [hunk 5]
```diff
@@ -39,28 +39,35 @@ def configs_from_dict(behavior_dict):
...
+        self.track_file_name = track_file_name
 
     def get_info_string_list(self):
         info_strings = ["ConfigIdx: {}".format(self.config_idx),
                         "Behavior: {}".format(self.behavior_config.behavior_name),
                         "ScenarioSet: {}".format(self.scenario_set_name),
                         "ScenarioIdx: {}".format(self.scenario_idx),
-                        "ScenarioParamDes: {}".format(self.scenario_set_param_desc)]
+                        "ScenarioParamDes: {}".format(self.scenario_set_param_desc),
+                        "TrackIdEgo: {}".format(self.track_id_ego),
```
Comment (juloberno) [inline bark/benchmark/benchmark_result.py:59]: See previous comment

File: bark/benchmark/benchmark_result.py [hunk 6]
```diff
@@ -39,28 +39,35 @@ def configs_from_dict(behavior_dict):
...
 
     def as_dict(self):
         return {"config_idx": self.config_idx,
                 "scen_set": self.scenario_set_name,
                 "scen_idx": self.scenario_idx,
                 **self.scenario_set_param_desc,
-                **self.behavior_config.as_dict()}
+                **self.behavior_config.as_dict(),
+                "track_id_ego": self.track_id_ego,
+                "track_file_name": self.track_file_name}
```
Comment (juloberno) [inline bark/benchmark/benchmark_result.py:70]: see first hints

File: bark/benchmark/benchmark_runner.py [hunk 7]
```diff
@@ -125,14 +125,20 @@ def _create_configurations(self, num_scenarios=None):
...
                     benchmark_config = \
                         BenchmarkConfig(
                             len(benchmark_configs),
                             behavior_config,
                             scenario,
                             scenario_idx,
                             scenario_set_name,
-                            scenario_set_param_desc
+                            scenario_set_param_desc,
+                            track_id_ego,
```
Comment (juloberno) [inline bark/benchmark/benchmark_runner.py:140]: see first hints

File: bark/benchmark/benchmark_runner.py [hunk 8]
```diff
@@ -289,3 +305,13 @@ def _log_eval_average(self, result_dct_list, configs):
         self.logger.info("\n------------------- Current Evaluation Results ---------------------- \n Num. Results:{}\n {} \n \
 ---------------------------------------------------------------------".format(len(result_dct_list),
                                                                               grouped.to_string()))
+
+    def _get_track_id_and_filename(self, scenario):
+        # only relevant for scenarios from dataset
```
Comment (juloberno) [inline bark/benchmark/benchmark_runner.py:310]: This should not be in benchmark runner (is not using self argument). Thus should be some free function somewhere in the interaction dataset scenario generation.

**Review by juloberno — CHANGES_REQUESTED** [review]

File: bark/benchmark/benchmark_result.py (hunk 6 above)
Comment (klemense1) [reply bark/benchmark/benchmark_result.py]: removed it

File: bark/benchmark/benchmark_result.py (hunk 5 above)
Comment (klemense1) [reply bark/benchmark/benchmark_result.py]: removed it

File: bark/benchmark/benchmark_result.py (hunk 4 above)
Comment (klemense1) [reply bark/benchmark/benchmark_result.py]: okay, I implemented it using scenario_set_param_desc

File: bark/benchmark/benchmark_runner.py (hunk 7 above)
Comment (klemense1) [reply bark/benchmark/benchmark_runner.py]: removed it

File: bark/benchmark/benchmark_runner.py (hunk 8 above)
Comment (klemense1) [reply bark/benchmark/benchmark_runner.py:309]: I don't think it makes sense to move it there, as I would not have control over it after being processed to the database. Essentially, the track file will change, as a benchmark will use multiple ones (that's why I want to log which one is used)

Comment (klemense1) [reply bark/benchmark/benchmark_runner.py]: I moved it to the scenario class

**Pull request merged**